Cookbook/Proposals/Comments.md

```diff
@@ -0,0 +1,224 @@
+                drawerState = .partiallyExpanded
+                case let .loaded(places):
+                drawerState = places.isEmpty.isFalse
+                 ? .partiallyExpanded
+                 : .collapsed
+                case .failed(.noPlaces):
+                 drawerState = .partiallyExpanded
+                  default:
+                 drawerState = (state.suggestions != nil)
+                 ? .partiallyExpanded
+                 : .collapsed
+        }
+        // Flow controls internal state of drawer presentation
+        // for example after it drawer is dismissed it shouldn't be possible
+        // to change its state without presenting it first.
+        // For that reason we change state through the flow
+        // and not through the self.drawerPresentationController.
+        // We can as well consider making this a default behaviour in DrawerKit
+        drawerFlow.setDrawerState(drawerState)
+}
+
+```
+
+
+ Ideally, it could be exported to `self?.setDrawerState(drawerState)` to not disturb a reader when it's not necessary:
+ 
+ ```swift
+    viewModel.state.signal
+      .observe(on: UIScheduler())
+      .observeValues { [weak self] state in
+        let drawerState: DrawerState
+        switch state.status {
+            case .searching:
+                drawerState = .fullyExpanded
+            case .loading:
+                drawerState = .partiallyExpanded
+            case let .loaded(places):
+                drawerState = places.isEmpty.isFalse
+                ? .partiallyExpanded
+                : .collapsed
+            case .failed(.noPlaces):
+                drawerState = .partiallyExpanded
+            default:
+                drawerState = (state.suggestions != nil)
+                ? .partiallyExpanded
+                : .collapsed
+        }
+        self?.setDrawerState(drawerState)
+    }
+
+// Flow controls internal state of drawer presentation
+// for example after it drawer is dismissed it shouldn't be possible
+// to change its state without presenting it first.        
+// For that reason we change state through the flow
+// and not through the self.drawerPresentationController.
+// We can as well consider making this a default behaviour in DrawerKit
+private func setDrawerState(_ drawerState: DrawerState) {
+    drawerFlow.setDrawerState(drawerState)
+}
+ ```
+ 
+
+
+## Impact on existing codebase
+It doesn't have any impact on existing codebase as this is about extending a rule in our StyleGuide. What will change is during code reviews we may have more suggestions on how to replace a comment with a code.
+
+
+## Alternatives considered
+1. Leave the StyleGuide as it is, aka. reject this proposal.
+2. Modify the list of good comments. Maybe we don't agree on everything in this proposal but we like part of it.
+3. We can also extend this proposal to define our approach to `MARK`. Currentely, `MARK` is not covered in the Style Guide neither in this proposal.
```

Comment:
I think it's worth having a discussion about inclusion of `// MARK`, but outside of this proposal, since this proposal is focused on explanatory comments, while considering using `// MARK` or not is more of a code-organisation matter and belongs in a separate discussion.

Reply:
I've already removed this line as I agree.
And I was originally wrong. The StyleGuide already defines our approach to `// MARK` ;)